CMakePresets.json

```diff
@@ -29,24 +21,64 @@
     {
       "name": "gcc-debug",
       "displayName": "GCC Debug Build",
+      "inherits": ["_root-config"],
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER": "g++",
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_CXX_FLAGS": "-fsanitize=address -fsanitize=pointer-compare -fsanitize=pointer-subtract -fsanitize=leak -fsanitize=undefined"
+      }
+    },
+    {
+      "name": "gcc-release",
+      "displayName": "GCC Release Build",
       "inherits": [
         "_root-config",
-        "_debug-base"
+        "_release-base"
       ],
       "cacheVariables": {
         "CMAKE_CXX_COMPILER": "g++"
       }
     },
     {
-      "name": "gcc-release",
-      "displayName": "GCC Release Build",
+      "name": "xcode-debug",
+      "displayName": "xcode Debug Build",
+      "inherits": ["_root-config"],
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER": "g++",
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_CXX_FLAGS": "-fsanitize=address -fsanitize=pointer-compare -fsanitize=pointer-subtract -fsanitize=undefined"
+      }
+    },
+    {
+      "name": "xcode-release",
+      "displayName": "xcode Release Build",
       "inherits": [
         "_root-config",
         "_release-base"
       ],
       "cacheVariables": {
         "CMAKE_CXX_COMPILER": "g++"
       }
+    },
+    {
+      "name": "msvc-debug",
+      "displayName": "MSVC Debug Build",
+      "inherits": ["_root-config"],
+      "cacheVariables": {
+        "CMAKE_CXX_COMPILER": "cl",
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_CXX_FLAGS": "/fsanitize=address /Zi"
```

Comment:
agree on `/permissive-`. Do we need to document default use of `EHsc` somewhere?

If `EHsc` is preferred, should we move this parameter to CMakefiles? this could be easily missed in e.g. CI scripts, it would make a lot of sense to be consistent on exception handling configuration across all build interfaces.

Reply:
Added `/permissive-`. Waiting for discussion on `/EHsc`.

Reply:
The rule is that the invoker of CMake sets compilation flags, not our `CMakeLists.txt` files. People will want to set the exception handling flag to what makes sense for their large project and we should not change that.

`cl.exe` documentation for EHsc makes it clear that this is the recommended setting. I don't think that this needs to be documented anywhere else. If people are curious, they can use git blame and see the rationale in this conversation.

Reply:
I will adapt with the assumption that this is industry norm.

Reply:
The industry norm is being really terrible at reuse and package management. If this were `rust`, `cargo` would take care of doing the right thing. Since it's C++, `cmake` gives you all the options and the wrong defaults.

Reply:
> The industry norm is being really terrible at reuse and package management. If this were `rust`, `cargo` would take care of doing the right thing. Since it's C++, `cmake` gives you all the options and the wrong defaults. 

It's not CMake but cl here. But generally yeah I agree...

Reply:
Adopted